Mathlib/Algebra/Module/LocalizedModule.lean

```diff
@@ -0,0 +1,1122 @@
+    mk' f m₁ s₁ - mk' f m₂ s₂ = mk' f (s₂ • m₁ - s₁ • m₂) (s₁ * s₂) := by
+  rw [sub_eq_add_neg, ← mk'_neg, mk'_add_mk', smul_neg, ← sub_eq_add_neg]
+#align is_localized_module.mk'_sub_mk' IsLocalizedModule.mk'_sub_mk'
+
+theorem mk'_mul_mk'_of_map_mul {M M' : Type _} [Semiring M] [Semiring M'] [Module R M]
+    [Algebra R M'] (f : M →ₗ[R] M') (hf : ∀ m₁ m₂, f (m₁ * m₂) = f m₁ * f m₂)
+    [IsLocalizedModule S f] (m₁ m₂ : M) (s₁ s₂ : S) :
+    mk' f m₁ s₁ * mk' f m₂ s₂ = mk' f (m₁ * m₂) (s₁ * s₂) := by
+  symm
+  apply (Module.End_algebraMap_isUnit_inv_apply_eq_iff _ _ _).mpr
+  simp_rw [Submonoid.coe_mul, ← smul_eq_mul]
+  rw [smul_smul_smul_comm, ← mk'_smul, ← mk'_smul]
+  simp_rw [← Submonoid.smul_def, mk'_cancel, smul_eq_mul, hf]
+#align is_localized_module.mk'_mul_mk'_of_map_mul IsLocalizedModule.mk'_mul_mk'_of_map_mul
+
+theorem mk'_mul_mk' {M M' : Type _} [Semiring M] [Semiring M'] [Algebra R M] [Algebra R M']
+    (f : M →ₐ[R] M') [IsLocalizedModule S f.toLinearMap] (m₁ m₂ : M) (s₁ s₂ : S) :
+    mk' f.toLinearMap m₁ s₁ * mk' f.toLinearMap m₂ s₂ = mk' f.toLinearMap (m₁ * m₂) (s₁ * s₂) :=
+  mk'_mul_mk'_of_map_mul f.toLinearMap f.map_mul m₁ m₂ s₁ s₂
+#align is_localized_module.mk'_mul_mk' IsLocalizedModule.mk'_mul_mk'
+
+variable {f}
+
+/-- Porting note: simp can prove this
+@[simp] -/
+theorem mk'_eq_iff {m : M} {s : S} {m' : M'} : mk' f m s = m' ↔ f m = s • m' := by
+  rw [← smul_inj f s, Submonoid.smul_def, ← mk'_smul, ← Submonoid.smul_def, mk'_cancel]
+#align is_localized_module.mk'_eq_iff IsLocalizedModule.mk'_eq_iff
+
+@[simp]
+theorem mk'_eq_zero {m : M} (s : S) : mk' f m s = 0 ↔ f m = 0 := by rw [mk'_eq_iff, smul_zero]
+#align is_localized_module.mk'_eq_zero IsLocalizedModule.mk'_eq_zero
+
+variable (f)
+
+theorem mk'_eq_zero' {m : M} (s : S) : mk' f m s = 0 ↔ ∃ s' : S, s' • m = 0 := by
+  simp_rw [← mk'_zero f (1 : S), mk'_eq_mk'_iff, smul_zero, one_smul, eq_comm]
+#align is_localized_module.mk'_eq_zero' IsLocalizedModule.mk'_eq_zero'
+
+theorem mk_eq_mk' (s : S) (m : M) :
+    LocalizedModule.mk m s = mk' (LocalizedModule.mkLinearMap S M) m s := by
+  rw [eq_comm, mk'_eq_iff, Submonoid.smul_def, LocalizedModule.smul'_mk, ← Submonoid.smul_def,
+    LocalizedModule.mk_cancel, LocalizedModule.mkLinearMap_apply]
+#align is_localized_module.mk_eq_mk' IsLocalizedModule.mk_eq_mk'
+
+variable (S)
+
+theorem eq_zero_iff {m : M} : f m = 0 ↔ ∃ s' : S, s' • m = 0 :=
+  (mk'_eq_zero (1 : S)).symm.trans (mk'_eq_zero' f _)
+#align is_localized_module.eq_zero_iff IsLocalizedModule.eq_zero_iff
+
+theorem mk'_surjective : Function.Surjective (Function.uncurry <| mk' f : M × S → M') := by
+  intro x
+  obtain ⟨⟨m, s⟩, e : s • x = f m⟩ := IsLocalizedModule.surj S f x
+  exact ⟨⟨m, s⟩, mk'_eq_iff.mpr e.symm⟩
+#align is_localized_module.mk'_surjective IsLocalizedModule.mk'_surjective
+
+section Algebra
+
+-- Porting note: Manually added since `inferInstance` fails.
+@[nolint docBlame]
```

Comment:
Please don't add `@[nolint docBlame]`.

Reply:
Perhaps this one is also worth investigating with #3414?

Reply:
As you guessed, merging #3414 fixed it!